frontend/simple/js/backend/hapi.js

```diff
@@ -73,29 +73,55 @@ export class HapiBackend extends Backend {
   subscriptions () {
     return Object.keys(store.state.contracts)
   }
-  async subscribe (contractId: string) {
+  async subscribeAndSyncIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.subscribeAndSync(identity, ephemeral)
+  }
+  async unsubscribeIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.unsubscribe(identity, ephemeral)
+  }
+  async subscribeAndSync (contractId: string, ephemeral: bool) {
+    await this.subscribe(contractId, ephemeral)
+    await store.dispatch('syncContractWithServer', contractId)
+  }
+  async subscribe (contractId: string, ephemeral: bool) {
     console.log('subscribing to:', contractId)
-    store.commit('pending', contractId)
-    // we don't need to check if we're already subscribed, server handles that
+    // Remove the ephemeral if we have decided to subscribe non-ephemerally
+    if (store.state.ephemerals[contractId] && !ephemeral) {
+      store.dispatch('removeEphemeral', contractId)
+      store.dispatch('removeContract', contractId)
```

Comment:
Why is the contract itself being removed? If there's a good reason, maybe a comment should be added to explain? (That's assuming literalizing ephemerals is a good idea in the first place)